overlord/devicestate/systems.go

```diff
@@ -426,3 +429,52 @@ func createSystemForModelFromValidatedSnaps(model *asserts.Model, label string,
 
 	return recoverySystemDir, nil
 }
+
+// ModelAndGadgetInfoFromSeed return the model assertion and gadget details
+// for the given system label.
+func ModelAndGadgetInfoFromSeed(wantedSystemLabel string) (*asserts.Model, *gadget.Info, error) {
+	s, err := seed.Open(dirs.SnapSeedDir, wantedSystemLabel)
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := s.LoadAssertions(nil, nil); err != nil {
+		return nil, nil, fmt.Errorf("cannot load assertions for label %q: %v", wantedSystemLabel, err)
+	}
+	model := s.Model()
+
+	// 2. get the gadget volumes for the given seed-label
+	perf := &timings.Timings{}
+	if err := s.LoadEssentialMeta([]snap.Type{snap.TypeGadget}, perf); err != nil {
+		return nil, nil, fmt.Errorf("cannot load gadget snap metadata: %v", err)
+	}
+	var gadgetSnapPath string
+	err = s.Iter(func(sn *seed.Snap) error {
+		gadgetSnapPath = sn.Path
+		return nil
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot read gadget snap: %v", err)
+	}
+	if gadgetSnapPath == "" {
+		return nil, nil, fmt.Errorf("internal error: cannot get gadget snap path")
+	}
+	snapf, err := snapfile.Open(gadgetSnapPath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot open gadget snap: %v", err)
+	}
+	gadgetYaml, err := snapf.ReadFile("meta/gadget.yaml")
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot read gadget.yaml: %v", err)
+	}
+	gadgetInfo, err := gadget.InfoFromGadgetYaml(gadgetYaml, model)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot parse gadget.yaml: %v", err)
+	}
+	// XXX: no opts for now
+	opts := &gadget.ValidationConstraints{}
```

Comment:
we need to look into this

Reply:
Indeed, I wonder if we need options that the installer can pass to indicate it wants encryption (or not)?

Reply:
more the reverse, as the spec hints at, we need to find out whether encryption is mandatory or preferred, etc and whether is possible in terms of hardware and gadget compatibility and add a field or fields about that info in the result